### PR TITLE
ADDED:Employee Management System

### PR DESCRIPTION
**Pull request by AKSHAYAPALCHAMY-P**

### CODE REVIEW

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/js/adminLogin.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+const ADMIN_EMAIL = document.getElementById("loginEmail");
+const ADMIN_PASSWORD = document.getElementById("loginPassword");
+
+AdminLoginMethods.Login = function()
+{
+    let strAdminEmail = ADMIN_EMAIL.value;
+    let strAdminPassword = ADMIN_PASSWORD.value;
+
+    let adminEmail = 'admin@gmail.com';
```
Comment (msindia-learning) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/js/adminLogin.js]: Change variable names as strOriginalAdminEmail

Comment (msindia-learning) [reply AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/js/adminLogin.js]: Follow the same for adminPassword

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/views/adminlogin.html
Comment (msindia-learning) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/views/adminlogin.html:1]: Change file name to AdminLogin.html

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/js/adminLogin.js
Comment (msindia-learning) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/js/adminLogin.js:1]: Change filename as AdminLogin.js

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/js/adminLogin.js (hunk 1 above)
Comment (AKSHAYAPALCHAMY-P) [reply AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/js/adminLogin.js]: done

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/views/adminlogin.html
Comment (AKSHAYAPALCHAMY-P) [reply AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/views/adminlogin.html:1]: done

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/js/AdminLogin.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+let AdminLoginMethods = {};
+
+
+const ADMIN_EMAIL = document.getElementById("loginEmail");
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/js/AdminLogin.js:4]: Good to see this naming.., 😊👌

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/js/adminLogin.js (hunk 1 above)
Comment (ManiRaj-021199) [reply AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/js/adminLogin.js]: Please check.. Not yet resolved..,

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Views/AdminLogin.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Admin-Login</title>
+</head>
+<body>
+
+    <div>
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Views/AdminLogin.html:10]: Where is the container div?

```
<div id="1"></div>
<div id="2"></div>

<div id="container">
  <div id="1"></div>
  <div id="2"></div>
</div>
```

Which one will you choose?

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
+let displayMethods = {};
+
+document.addEventListener("DOMContentLoaded", function()
+{
+
+    const saveButton = document.getElementById("saveButton");
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:6]: const naming rule - ALL_CAPS

SAVE_BUTTON

Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:6]: Move document.getElementById to global declaration.

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    const employeeDetailsDialog = document.getElementById("employeeDetailsDialog");
+    const okButton = document.getElementById("okButton");
+
+    saveButton.addEventListener("click", function()
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:10]: Why addEventListener -> use onclick instead

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    {
+
+        const name = document.getElementById('EmpName').value;
+        const age = document.getElementById('EmpAge').value;
+        const address = document.getElementById('EmpAddress').value;
+        const qualification = document.getElementById('EmpQualification').value;
+        const position = document.getElementById('EmpPosition').value;
+        const salary = document.getElementById('Empsalary').value;
+        const genderElement = document.querySelector('input[name="gender"]:checked');
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:19]: move all const to global.

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+
+
+        //if (!genderElement) {
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:23]: remove commented codes

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+            "Employee Name": name,
+            "Employee Age": age,
+            "Employee Address": address,
+            "Employee Qualification": qualification,
+            "Employee Position": position,
+            "Employee Salary": salary,
+            "Gender": gender
+        };
+
+        let details = JSON.parse(localStorage.getItem("Details")) || [];
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:40]: Why everywhere getting and parsing localstorage values? Think about alternative solutions? 😊

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js [hunk 9]
```diff
@@ -0,0 +1,136 @@
+let displayMethods = {};
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:1]: EmployeeDetailMethods

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js (hunk 4 above)
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:6]: Give meaningful name - AddEmployee

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+            "Gender": gender
+        };
+
+        let details = JSON.parse(localStorage.getItem("Details")) || [];
+
+        details.push(employeeDetails);
+
+        localStorage.setItem("Details", JSON.stringify(details));
+
+        document.getElementById("employeeForm").reset();
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:46]: Nice 😊

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+        const employeeDetailsContent = document.getElementById("employeeDetailsContent");
+        employeeDetailsContent.innerHTML = "";
+
+        if(details.length > 0)
+        {
+            const detailsHTML = "<h2>Employee Details</h2>" +
+                "<ul>" +
+                details.map((detail, index) =>
+                {
+                    return `<li>${detail["Employee Name"]} - ${detail["Employee Age"]} - ${detail["Employee Address"]} - ${detail["Employee Qualification"]} - ${detail["Employee Position"]} - ${detail["Employee Salary"]} - ${detail["Gender"]}</li><button onclick="displayMethods.editEmployee(${index})">Edit</button> <button onclick="displayMethods.deleteEmployee(${index})">Delete</button></li>`;
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:70]: Hardcoded string.. Any other ways to achieve this?

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+
+
+
+
+    displayMethods.editEmployee = function(index)
+    {
+        let details = JSON.parse(localStorage.getItem("Details")) || [];
+        const employee = details[index];
+
+        document.getElementById('EmpName').value = employee["Employee Name"];
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:88]: Duplicate code. Use global value instead.

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+        document.getElementById('EmpAddress').value = employee["Employee Address"];
+        document.getElementById('EmpQualification').value = employee["Employee Qualification"];
+        document.getElementById('EmpPosition').value = employee["Employee Position"];
+        document.getElementById('Empsalary').value = employee["Employee Salary"];
+        document.querySelector(`input[name="gender"][value="${employee["Gender"]}"]`).checked = true;
+
+        saveButton.innerHTML = "Update";
+        saveButton.removeEventListener("click", saveNewEmployee);
+        saveButton.addEventListener("click", function()
+        {
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:99]: This implementation looks nice. 😊

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+        details[index] = {
+            "Employee Name": document.getElementById('EmpName').value,
+            "Employee Age": document.getElementById('EmpAge').value,
+            "Employee Address": document.getElementById('EmpAddress').value,
+            "Employee Qualification": document.getElementById('EmpQualification').value,
+            "Employee Position": document.getElementById('EmpPosition').value,
+            "Employee Salary": document.getElementById('EmpSalary').value,
+            "Gender": document.querySelector('input[name="gender":checked]').value,
+        }
+        localStorage.setItem("Details", JSON.stringify(details));
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:118]: Give proper name - Details -> EmployeeDetails

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+
+        document.getElementById("employeeForm").reset();
+        saveButton.innerHTML = "Save";
+        saveButton.removeEventListener("click", displayMethods.updateEmployee);
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:122]: Move this logic to separate function

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js [hunk 16]
```diff
@@ -0,0 +1,136 @@
+let displayMethods = {};
+
+document.addEventListener("DOMContentLoaded", function()
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/JS/EmployeeDetails.js:3]: All methods are inside DOMContentLoaded -> Wrong approach.
Just call the function which is require to execute after page rendered..,

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/js/AdminLogin.js [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+{
+    let strAdminEmail = ADMIN_EMAIL.value;
+    let strAdminPassword = ADMIN_PASSWORD.value;
+
+    let strAdminOriginalEmail = 'admin@gmail.com';
+    let strAdminOriginalPassword = 'password';
+
+    if(strAdminEmail === strAdminOriginalEmail && strAdminPassword === strAdminOriginalPassword)
+    {
+        alert("Login Successful!");
```
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Assets/js/AdminLogin.js:17]: Do not use hardcoded string

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/views/adminlogin.html
Comment (ManiRaj-021199) [inline AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/views/adminlogin.html:1]: Change file name - AdminLogin.html

**Review by ManiRaj-021199 — CHANGES_REQUESTED** [review]
Please have a look

File: AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Views/AdminLogin.html (hunk 3 above)
Comment (AKSHAYAPALCHAMY-P) [reply AkshayaP/00 - Projects/04-Frontend Projects/Employee Management System/Views/AdminLogin.html:10]: 2 nd one.

**Comment by ManiRaj-021199** [comment]
Files not updated properly.

**Pull request closed without merging**